plugins/janus_videoroom.c

```diff
@@ -6825,11 +6895,14 @@ static gboolean janus_gst_create_pipeline(forward_media_type media_type,
 		break;
 	case MEDIA_AUDIO_MIXER:
 		gstr = &room->gst_thread_parameters[MEDIA_AUDIO_MIXER].gstr;
-
+		if(dynamic_url_status){
+			g_snprintf(rtsp_url_base,JANUS_RTP_FORWARD_STRING_SIZE,"rtsp://%s:%s@%s:1935/ClientVideo/", room->sgwUsername, room->sgwPassword, room->sgwURL);
+		}
+		
 		IS_PARAM_IN_LIMITS(g_snprintf(log_string, MAX_STRING_LEN, "AUDIO MIXER %s", room->room_id_str),
 						   "log_string", 0, MAX_STRING_LEN);
 
-		IS_PARAM_IN_LIMITS(g_snprintf(rtsp_full_url, JANUS_RTP_FORWARD_STRING_SIZE, "%sAUDIO_%s", rtsp_url, room->room_id_str),
+		IS_PARAM_IN_LIMITS(g_snprintf(rtsp_full_url, JANUS_RTP_FORWARD_STRING_SIZE, "%sAUDIO_%s", dynamic_url_status ? rtsp_url_base : rtsp_url, room->room_id_str),
```

Comment:
Please try to build the URL beforehand depending on the dynamuc_url_status and pass the URL accorindgly here without asking the variable, only for making it more cleaner.

Reply:
what do you mean , here the url is already constructed , other from config or from http request but is being added the VIDEO_ID or AUDIO_ID

Reply:
fixed